src/parser/job.rs

```diff
@@ -84,7 +113,97 @@ impl std::cmp::PartialEq for Job {
 impl Job {
     pub fn start(self: &mut Self, job_name: &String) {
         println!("log: start {}", job_name);
-        self.is_running = true;
+
+        for i in 0..self.num_procs {
+            let mut command = Command::new(&self.command);
+
+            if let Some(args) = &self.arguments {
+                for element in args.iter() {
+                    command.arg(element);
+                }
+            }
+
+            if let Some(child) = self.processes.get_mut(&i) {
+                match child.try_wait() {
+                    Ok(None) => {
+                        println!("Process is already running.");
+                        continue;
+                    }
+                    Ok(Some(_)) => {}
+                    Err(e) => {
+                        eprintln!("Error: {:?}", e);
+                        return;
+                    }
+                }
+            }
+
+            if let Some(environment) = &self.environment {
+                for (key, value) in environment {
+                    command.env(key, value);
+                }
+            }
+
+            // if let Some(ref config_umask) = self.umask {
+            //     match command.pre_exec( || {
+            //         unsafe { libc::umask(config_umask) }; // checker si c'est le bon input
+            //     }) {
+            //         Ok(_) => {}
+            //         Err(e) => {
+            //             eprintln!("Error: {:?}", e);
+            //             return;
+            //         }
+            //     }
+            // }
+
+            if let Some(ref work_dir) = self.work_dir {
+                if fs::metadata(work_dir).is_err() {
+                    eprintln!("Error: {:?}", work_dir);
```

Comment:
https://doc.rust-lang.org/std/path/struct.Path.html#method.is_dir 
use is_dir to check if the path is a directory